goon/code/obj/machinery/bot/chefbot.dm

```diff
@@ -0,0 +1,195 @@
+	anchored = 0
+	on = 1 // ACTION
+	health = 50
+	var/raging = 0
+	var/list/calledout = list()
+	var/shit_reagents = list(TOXIN,STOXIN,SLIMEJELLY,PLASMA,NANITES,HONKSERUM,SILENCER,DISCOUNT,AMATOXIN,PSILOCYBIN,CARPPHEROMONES,BUSTANUT,ROGAN,MOONROCKS,TOXICWASTE,CHEMICAL_WASTE,HORSEMEAT,OFFCOLORCHEESE,BONEMARROW,IRRADIATEDBEANS,MUTATEDBEANS,MINTTOXIN,MERCURY,MINDBREAKER,SPIRITBREAKER,RADIUM,URANIUM,CARPOTOXIN,ZOMBIEPOWDER,AMUTATIONTOXIN,PACID,CHLORALHYDRATE,LITHIUM,HELL_RAMEN)
+
+/obj/machinery/bot/chefbot/proc/do_step()
+	var/turf/moveto = locate(src.x + rand(-1,1),src.y + rand(-1, 1),src.z)
+	if(isturf(moveto) && !moveto.density) step_towards(src, moveto)
+
+/obj/machinery/bot/chefbot/New()
+	..()
+	drama()
+
+/obj/machinery/bot/chefbot/process()
+	if(raging)
+		return
+	if(prob(60) && src.on == 1)
+		spawn(0)
+			do_step()
+			if(prob(30 + src.emagged * 30))
+				yell()
+
+/obj/machinery/bot/chefbot/proc/point(var/target)
+	visible_message("<b>[src]</b> points at [target].")
+	if(istype(target, /atom))
+		var/D = new /obj/effect/decal/point(get_turf(target))
+		spawn(25)
+			qdel(D)
+
+/obj/machinery/bot/chefbot/proc/drama()
+	for(var/mob/M in hearers(7, src))
+		M << sound('goon/sound/effects/dramatic.ogg', volume = 100) // F U C K temporary measure
+
+/obj/machinery/bot/chefbot/proc/why_is_it_bad()
+	return pick("IS FUCKING [pick("RAW", "BLAND", "UNDERCOOKED", "OVERCOOKED", "INEDIBLE", "RANCID", "DISGUSTING", "GARLICKY", "INEDIBLE", "SALTY", "NOT GOOD ENOUGH", "GREASY")]", "LOOKS LIKE [pick("IT'S NOT EVEN COOKED", "BABY VOMIT", "A MUSHY PIG'S ASS", "REGURGITATED DONKEY SHIT", "A PILE OF ROTTING FLIES", "REFINED CATBEAST PISS", "DOG DINNER", "ANEMIC BITS OF SHIT")]")
+
+/obj/machinery/bot/chefbot/proc/yell()
+	if(prob(50))
+		var/obj/item/weapon/reagent_containers/food/snacks/shitfood
+		var/mob/living/carbon/human/thechef
+		var/mob/dork
+		var/is_thechef_the_chef = 0
+		for(var/obj/item/weapon/reagent_containers/food/snacks/probablyshitfood in view(7, src))
+			if(probablyshitfood in calledout)
+				continue
+			if(probablyshitfood.reagents)
+				var/datum/reagents/R = probablyshitfood.reagents
+				if((R.has_any_reagents(shit_reagents) || R.has_reagent(CORNOIL, 10)) && !emagged) // FUCK OFF ARE YOU TRYING TO KILL SOMEONE? || THIS SHIT IS SO GREASY IT COULD BE USED TO FUEL A ESCAPE SHUTTLE TRIP TO BACK TO CENTOMM
+					shitfood = probablyshitfood
+				if(!R.has_any_reagents(shit_reagents) && emagged)
+					shitfood = probablyshitfood
+					break
+		if(shitfood)
+			raging = 1
+			update_icon()
+			for(var/mob/living/carbon/human/M in view(7, src))
+				if(M.mind)
+					if(M.mind.assigned_role == "Chef")
+						thechef = M
+						is_thechef_the_chef = 1
+						break
+				if(M.wear_id)
+					if(findtext(M.wear_id:assignment, "chef") || findtext(M.wear_id:assignment, "cook"))
+						thechef = M
+						is_thechef_the_chef = 1
+						break
+				if(!thechef)
+					thechef = M
+				if(!dork)
+					if(M.client)
+						if(M.client.IsByondMember())
```

Comment:
well, it's not really an incentive, the only thing that it is used for is to determine if he can ask if they "bought their [cooking] license"

seems harmless